drivers/hmis/app/models/hmis/hud/validators/enrollment_validator.rb

```diff
@@ -110,7 +110,13 @@ def self.hmis_validate(record, role: nil, **_)
 
   def validate(record)
     super(record) do
-      record.errors.add :project_id, :required if record.project_id.nil? && record.wip.nil?
+      if record.project
+        # enrollment.project_id should match the project.project_id. The rails association is through project_pk
+        record.errors.add :project_id, 'reference does not match DB PK' if record.ProjectID && record.ProjectID != record.project.ProjectID
+        record.errors.add :project_id, 'must match enrollment data source' if record.project.data_source_id != record.data_source_id
+      else
+        record.errors.add :project_id, 'must be present'
```

Comment:
use the default message for a required field
```suggestion
        record.errors.add :project_id, :required
```

Reply:
although kinda confused about this part, we would enter this case if `project_pk` is nil or invalid. So really the error message should say that `project_pk` is required, right?

Reply:
good catch, I'll change it to use project_pk. I just wanted something to validate that the project was present